__tests__/extension-requests/extension-requests.test.js

```diff
@@ -919,4 +939,44 @@ describe('Tests the Extension Requests Screen', () => {
     );
     expect(ascSortIconDisplayStyle).toBe('block');
   });
+
+  it('Should show empty message if all extension requests have been addressed', async () => {
+    await page.goto(`${baseUrl}/?order=asc&size=1&q=status%3APENDING`);
+    await page.waitForNetworkIdle();
+
+    extensionRequestsElement = await page.$('.extension-requests');
+    let extensionCardsList = await page.$$('.extension-card');
+
+    if (extensionCardsList.length == 0) {
+      const extensionRequestContainerText = await page.evaluate(
+        (element) => element.innerText,
+        extensionRequestsElement,
+      );
+
+      expect(extensionRequestContainerText).toBe(
+        'No extension requests to show!',
+      );
+
+      return;
+    }
+
+    for (const card of extensionCardsList) {
+      let approveButton = await card.$('.approve-button');
+      await approveButton.click();
+      await page.waitForTimeout(1700);
+    }
+
+    extensionRequestsElement = await page.$('.extension-requests');
+    extensionCardsList = await extensionRequestsElement.$$('.extension-card');
+
+    const extensionRequestContainerText = await page.evaluate(
+      (element) => element.innerText,
+      extensionRequestsElement,
+    );
+
+    expect(extensionCardsList.length).toBe(0);
+    expect(extensionRequestContainerText).toBe(
+      'No extension requests to show!',
+    );
+  });
```

Comment:
Which check will be tested here?

first one or the second one, how does it works?

Reply:
Both checks will be tested. 

The first one will be checked when the API fetches 0 extension requests (no extension requests remaining).

The second one will be checked when the API fetches 1 or more extension requests.